language/stdlib/modules/DiemAccount.move

```diff
@@ -1627,6 +1638,24 @@ module DiemAccount {
             Errors::invalid_argument(PROLOGUE_EINVALID_ACCOUNT_AUTH_KEY),
         );
 
+
+        let num_secondary_signers = Vector::length(&secondary_addresses);
+        assert(
+            Vector::length(&secondary_public_keys) == num_secondary_signers,
+            Errors::invalid_argument(PROLOGUE_EBAD_SECONDARY_PUB_KEYS),
+        );
+
+        let i = 0;
+        while (i < num_secondary_signers) {
```

Comment:
Should we put a conservative cap on the number of secondary signers for starters (8, to pull a # out of nowhere?). I'm not sure if we have compelling use-cases for >3 yet, and there's definitely a larger signature checking cost for more signers (especially given that each signer can have a k-of-n multisig account). CC @kchalkias for opinions on this too.

Reply:
Good catch, we have a few options here. We can either limit the num of signers or we could have a function over the total num of keys involved. I prefer the less error prone on num of signers as @sblackshear mentioned. A multisig key per account (k of N) can now reach up to N = 32 public key. 

Benchmarks:
My MacBook Pro i7 benches  show ~50 microsec per Ed25519 signature verification. If we allow 8 signers in total (1 major + 7 secondary) we're at 256 signatures in the worst case, which will cost ~13ms. Transaction size will also be 256 x (pubkeys (32bytes each) + sigs (64bytes each)) = 24.5K, are we ok with this?

Reply:
Both of those seem reasonable to me, though we should check with some Diem Core perf experts too. I suppose we could also look into parallel signature verification at the tx level if this becomes a problem.

Reply:
One thing that Emma suggested which may also be useful here is if we compute the `sha3_256` hash during the parallel signature checking of transactions that we're already performing, and then we only need to do equality comparisons here. Not sure how much of a win this might be, but something to consider when thinking about this bound.